mesonbuild/dependencies/dub.py

```diff
@@ -300,12 +309,106 @@ def find_package_target(pkg: T.Dict[str, str]) -> bool:
     # compiler, architecture, configuration...
     # It returns (target|None, {compatibilities})
     # If None is returned for target, compatibilities will list what other targets were found without full compatibility
-    def _find_compatible_package_target(self, jdesc: T.Dict[str, str], jpack: T.Dict[str, str], dub_comp_id: str) -> T.Tuple[str, T.Set[str]]:
-        dub_build_path = os.path.join(jpack['path'], '.dub', 'build')
+    def _find_target_in_cache(self, jdesc: JsonDict, jpack: JsonDict, dub_comp_id: str) -> T.Tuple[str, T.Set[str]]:
+
+        # platform, arch and compiler_version are arrays
+        spec: JsonDict = {
+            'name': jpack['name'],
+            'version': jpack['version'],
+            'path': jpack['path'],
+            'target_file_name': jpack['targetFileName'],
+            'configuration': jpack['configuration'],
+            'build_type': jdesc['buildType'],
+            'platform': jdesc['platform'],
+            'arch': jdesc['architecture'],
+            'compiler': dub_comp_id,
+            'compiler_version': self._get_comp_versions_to_find(dub_comp_id),
+        }
+
+        # build_type is not in check_list because different build types might be compatible.
+        # We do show a WARNING that the build type is not the same.
+        # It might be critical in release builds, and acceptable otherwise
+        check_list = {'configuration', 'platform', 'arch', 'compiler', 'compiler_version'}
+
+        mlog.debug('Searching in DUB cache for compatible', spec['target_file_name'])
+
+        if self.use_cache_db:
+            result = self._find_in_cache_db(spec, check_list)
+        else:
+            result = self._find_in_old_cache(spec, check_list)
```

Comment:
... and make these yield `entries` instead of `result`.

Reply:
The downside of doing that is that I'll have to join the items in the cache DB in a single string to match the format of what I get from old cache.